proposals/p3848.md

```diff
@@ -0,0 +1,587 @@
+or positional parameters.
+
+The presence of "`->` _return-type_" determines whether the function body can
+(and must) return a value.
+
+To understand how the syntax between lambdas and function declarations is
+reasonably "continuous", refer to this table of syntactic positions and the
+following code examples.
+
+| Syntactic Position |                        Syntax Allowed in Given Position (optional, unless otherwise stated)                        |
+| :----------------: | :----------------------------------------------------------------------------------------------------------------: |
+|         A1         |               Required Returned Expression ([positional parameters](#positional-parameters) allowed)               |
+|         A2         |             Required Returned Expression ([positional parameters](#positional-parameters) disallowed)              |
+|         B          |                                   [Default Capture Mode](#default-capture-mode)                                    |
+|         C          | Explicit [Captures](#function-captures), [Function Fields](#function-fields) and Deduced Parameters (in any order) |
+|         D          |                                                Explicit Parameters                                                 |
+|         E1         |           Body of Statements (no return value) ([positional parameters](#positional-parameters) allowed)           |
+|         E2         |          Body of Statements (with return value) ([positional parameters](#positional-parameters) allowed)          |
+|         E3         |         Body of Statements (no return value) ([positional parameters](#positional-parameters) disallowed)          |
+|         E4         |        Body of Statements (with return value) ([positional parameters](#positional-parameters) disallowed)         |
+|         F          |                                                Required Return Type                                                |
+|         G          |                                             Function Declaration Name                                              |
+
+```
+// Lambdas (all the following are in an expression context and are
+// themselves expressions)
+
+fn => A1
+
+fn[B, C] => A1
+
+fn(D) => A2
+
+fn[B, C](D) => A2
+
+fn { E1; }
+
+fn -> F { E2; }
+
+fn[B, C] { E1; }
+
+fn[B, C] -> F { E2; }
+
+fn(D) { E3; }
+
+fn(D) -> F { E4; }
+
+fn[B, C](D) { E3; }
+
+fn[B, C](D) -> F { E4; }
+```
+
+```
+// Function Declarations (all the following are allowed as statements in a
+// function body or as declarations in other scopes)
+
+fn G => A1;
+
+fn G[B, C] => A1;
+
+fn G(D) => A2;
+
+fn G[B, C](D) => A2;
+
+fn G { E1; }
+
+fn G -> F { E2; }
+
+fn G[B, C] { E1; }
+
+fn G[B, C] -> F { E2; }
+
+fn G(D) { E3; }
+
+fn G(D) -> F { E4; }
+
+fn G[B, C](D) { E3; }
+
+fn G[B, C](D) -> F { E4; }
+```
+
+### Alternative Considered: Terse vs Elaborated
```

Comment:
The alternatives considered are spread throughout the proposal rather than grouped at the bottom which I think makes it a little bit harder to get a sense of what the actual proposal is.

But maybe more importantly, it would be good to try to summarize the pros, cons, and what tipped the scales to not pursue the alternative for each one. That's an important part of the structure we typically want in proposals.